### PR TITLE
Counter example

### PR DESCRIPTION
**Pull request by WillemHoman**
Simple example to demonstrate basics of implementing a builder and model

https://user-images.githubusercontent.com/2238749/136505460-a0a1592b-dad1-425f-a5af-805cf4ff5615.mov

@hx

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /node_modules
 /dist
+.idea
```
Comment (hx) [inline .gitignore:3]: This is a local environment [thing that you can do for your whole machine](https://stackoverflow.com/questions/7335420/global-git-ignore).

File: src/index.ts [hunk 2]
```diff
@@ -5,13 +5,17 @@ import { GameView } from './examples/chess/GameView'
 import { createOrderBuilder } from './examples/order/builder'
 import { OrderView } from './examples/order/OrderView'
 import { Welcome, createNullBuilder } from './examples/welcome'
+import { CounterView } from './examples/counter/CounterView'
+import { createCounterBuilder } from './examples/counter/builder'
```
Comment (hx) [inline src/index.ts:9]: Please add `examples/counter/index.ts`.

I should have done this for `examples/order` as well but I was in a bit of a rush.

File: src/examples/counter/Counter.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface Counter {
+  count: number
+}
```
Comment (hx) [inline src/examples/counter/Counter.ts:3]: What do you think about making this a simple numeric type instead? I see it as a good opportunity to demonstrate that models can be _anything_, not just objects with properties.

```suggestion
type Counter = number
```

You could simplify further by removing the `Counter` type completely and just using `number` directly.

**Review by hx — CHANGES_REQUESTED** [review]
Thanks @WillemHoman for this nice addition!

Any particular reason to remove `package-lock.json`? I would imagine from my work with gems that you would exclude them only from libraries intended for use in other projects, which this project is not.

**Comment by WillemHoman** [comment]
> Any particular reason to remove `package-lock.json`? 
It hasn't been removed, if you expand it in file view there are a couple of changes.

I think I deleted `package-lock.json` at one point and subsequently did an `npm install` which made this new  `package-lock.json`.

> I would imagine from my work with gems that you would exclude them only from libraries intended for use in other projects, which this project is not.
Agreed

**Pull request closed without merging**